velox/vector/LazyVector.cpp

```diff
@@ -82,4 +83,87 @@ void VectorLoader::loadInternal(
   load(positions, hook, result);
 }
 
+//   static
+void LazyVector::ensureLoadedRows(
+    VectorPtr& vector,
+    const SelectivityVector& rows) {
+  if (isLazyNotLoaded(*vector)) {
+    DecodedVector decoded;
+    SelectivityVector baseRows;
+    ensureLoadedRows(vector, rows, decoded, baseRows);
+  }
+}
+
+// static
+void LazyVector::ensureLoadedRows(
+    VectorPtr& vector,
+    const SelectivityVector& rows,
+    DecodedVector& decoded,
+    SelectivityVector& baseRows) {
+  decoded.decode(*vector, rows, false);
+  VELOX_CHECK_EQ(decoded.base()->encoding(), VectorEncoding::Simple::LAZY);
```

Comment:
This check makes it not safe to call this without without isLazyNotLoaded check, but the document suggests it would be safe to do. Perhaps, replace with if + early return.